platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/events/UnsignedEvent.java

```diff
@@ -161,53 +169,158 @@ public BaseEventHashedData(
+     * @return the deserialized event
+     * @throws IOException if unsupported payload types are encountered
+     */
+    public static UnsignedEvent deserialize(final SerializableDataInputStream in) throws IOException {
+        final int version = in.readInt();
+        if (version != ClassVersion.BIRTH_ROUND) {
+            throw new IOException("Unsupported version: " + version);
+        }
+
         Objects.requireNonNull(in, "The input stream must not be null");
-        softwareVersion = in.readSerializable(StaticSoftwareVersion.getSoftwareVersionClassIdSet());
+        final SoftwareVersion softwareVersion =
+                in.readSerializable(StaticSoftwareVersion.getSoftwareVersionClassIdSet());
 
-        creatorId = in.readSerializable(false, NodeId::new);
+        final var creatorId = in.readSerializable(false, NodeId::new);
         if (creatorId == null) {
             throw new IOException("creatorId is null");
         }
-        selfParent = in.readSerializable(false, EventDescriptor::new);
-        otherParents = in.readSerializableList(AddressBook.MAX_ADDRESSES, false, EventDescriptor::new);
-        allParents = createAllParentsList();
-        birthRound = in.readLong();
+        final var selfParent = in.readSerializable(false, EventDescriptor::new);
+        final var otherParents = in.readSerializableList(AddressBook.MAX_ADDRESSES, false, EventDescriptor::new);
+        final var birthRound = in.readLong();
 
-        timeCreated = in.readInstant();
+        final var timeCreated = in.readInstant();
         in.readInt(); // read serialized length
-        transactions =
-                in.readSerializableArray(ConsensusTransactionImpl[]::new, maxTransactionCount, true, TRANSACTION_TYPES);
+        final List<OneOf<PayloadOneOfType>> transactionList = new ArrayList<>();
+        final int payloadSize = in.readInt();
+        if (payloadSize > 0) {
+            in.readBoolean(); // allSameClass
+        }
+        for (int i = 0; i < payloadSize; i++) {
+            final long classId = in.readLong();
+            final int classVersion = in.readInt();
+            if (classId == APPLICATION_TRANSACTION_CLASS_ID) {
+                transactionList.add(new OneOf<>(
+                        PayloadOneOfType.APPLICATION_PAYLOAD, deserializeApplicationPayload(in, classVersion)));
+            } else if (classId == STATE_SIGNATURE_CLASS_ID) {
+                transactionList.add(new OneOf<>(
+                        PayloadOneOfType.STATE_SIGNATURE_PAYLOAD, deserializeStateSignaturePayload(in, classVersion)));
+            } else {
+                throw new IOException("Unknown classId: " + classId);
+            }
+        }
+
+        return new UnsignedEvent(
+                softwareVersion, creatorId, selfParent, otherParents, birthRound, timeCreated, transactionList);
+    }
+
+    private static Bytes deserializeApplicationPayload(SerializableDataInputStream in, int classVersion)
```

Comment:
params missing nullity and `final`. Return value missing nullity

Reply:
done